### PR TITLE
:bug: do not throw an exception when no provider is used

### LINKED ISSUE #73: Throw a readable exception when using inject in a react tree without provider
For now, when you forgot to use the `provider` HOC in your application, it throws cryptic error :

image

I suggest to have a more cleaner exception with a hint for the developer and a link to the documentation.

### PR DESCRIPTION
**Pull request by EmrysMyrddin**
fixes #73

### CODE REVIEW

**Comment by EmrysMyrddin** [comment]
@fabienjuif I'm a bit stuck with the tests. Do you have some advices on this ?

File: packages/connectors/react/misc/test-suite.js [hunk 1]
```diff
@@ -10,11 +10,7 @@ import { mount } from 'enzyme'
 import { createStore } from 'k-ramel'
 
 export default (lib) => {
-  const {
-    provider,
-    inject,
-    listen,
-  } = lib
+  const { provider, inject, listen } = lib
```
Comment (fabienjuif) [inline packages/connectors/react/misc/test-suite.js]: Can you desactivate your prettier ?

**Comment by fabienjuif** [comment]
@EmrysMyrddin for tests, you have to build project, I run test on both sources and build version to make sure I don't forget to build before publishing to NPM :)

in the root of project: `yarn build && yarn test`

Seems fine here

File: packages/connectors/react/misc/test-suite.js (hunk 1 above)
Comment (EmrysMyrddin) [reply packages/connectors/react/misc/test-suite.js]: It's already desactivated. This is caused by `eslint --fix` I think... It seems that the test files are not linted by the CI

**Comment by EmrysMyrddin** [comment]
Sorry for the rebase but there was huge conflicts problems when merging master.

**Comment by EmrysMyrddin** [comment]
@fabienjuif This is ready for review.

File: packages/connectors/react/src/inject.jsx [hunk 2]
```diff
@@ -31,8 +31,18 @@ export default injectFunction => WrappedComponent => class extends Component {
   }
 
   componentWillMount() {
+    const { store } = this.context
+
+    if (!store) {
+      // eslint-disable-next-line no-console
+      console.error('To use the inject function, one of the parents of this component needs to be a `provider`.\n' +
```
Comment (fabienjuif) [inline packages/connectors/react/src/inject.jsx]: I like the idea of :
 - Show what is the component that need to be fixed
 - Give a link to the documentation
 - Give an hint (use `provider`)

I have the feeling that:
 - we need to say "hey! The error comes from a k-ramel component!"
 - this is not quite exact to say a parents should be a `provider` (the important part is that the store is in the context, you can use our HoC or a Component to provide this, like `Provider` from `react-redux`).

What do you think about this?
```
[@k-ramel/react] Error in `inject` the Component for onlyUpdateForPropTypes(Header)

The store needs to be provided by an ancestor of this component.
> You can use `provider` from `@k-ramel/react` or `Provider` from react-redux.

Check the documentation for an example at https://github.com/alakarteio/k-ramel#connect-it-with-reactjs
```

Comment (EmrysMyrddin) [reply packages/connectors/react/src/inject.jsx]: I'm agree, it's better like this.

Comment (fabienjuif) [reply packages/connectors/react/src/inject.jsx]:
```
[@k-ramel/react] Error while wrapping `inject` to the Component onlyUpdateForPropTypes(Header)
/// ....
 ```

**Comment by EmrysMyrddin** [comment]
If we have more errors like this to log, we should probably define a utility function to help for styling/formating and adding documentation links.

Perhaps a lib already exists to handle log styling in a cleaner way ?

**Comment by fabienjuif** [comment]
This is not worth imo to add an already log system.

The maintainer that would add a log should factorise code by himself.

**Comment by fabienjuif** [comment]
Thank you @EmrysMyrddin <3

**Pull request closed without merging**